Objects/longobject.c

```diff
@@ -24,6 +24,9 @@ class int "PyObject *" "&PyLong_Type"
 _Py_IDENTIFIER(little);
 _Py_IDENTIFIER(big);
 
+static PyLongObject *DIGIT_ONE = &_PyLong_SMALL_INTS[_PY_NSMALLNEGINTS + 1];
+static PyLongObject *DIGIT_TWO = &_PyLong_SMALL_INTS[_PY_NSMALLNEGINTS + 2];
```

Comment:
Uses of `DIGIT_ONE` should be replaced with uses of `_PyLong_GetOne`. You may want to write a new `_PyLong_GetTwo` function along the same lines to replace uses of `DIGIT_TWO`. At the same time, the check `#if _PY_NSMALLPOSINTS < 2` in `pycore_long.h` would need to be updated to replace the `< 2` with `< 3`. (It's supposed to be possible to build CPython with different choices of `_PY_NSMALLNEGINTS` and `_PY_NSMALLPOSINTS` than the default ones, so it's not a priori safe to assume that `2` is a small integer.)

Reply:
@markshannon reverted my change which made int singletons per interpreter: https://bugs.python.org/issue45691#msg405514

I'm working on https://bugs.python.org/issue46417 to fix the root issue, to make sure that Py_Finalize()+Py_Initialize() remains safe.

Reply:
Done.